tools/docker/Dockerfile

```diff
@@ -0,0 +1,184 @@
+# syntax=docker/dockerfile:experimental
+
+# Common tools needed for the build stages ahead. The final end-use images (that we run tests in) are not derived from
+# this image, so the installs here will not take up space in those final images.
+FROM debian:stretch as build-base
+ENV PATH /bin:/usr/bin:/usr/local/bin:/sbin:/usr/sbin:/usr/local/sbin:/usr/lib/go/bin:/root/.cargo/bin
+RUN --mount=type=cache,id=aptcache,target=/var/cache/apt --mount=type=cache,id=aptlib,target=/var/lib/apt \
+mkdir -p /etc/apt/sources.list.d \
+&& echo "deb http://ftp.debian.org/debian stretch-backports main" > /etc/apt/sources.list.d/stretch-backports.list \
+&& apt update \
+&& apt --target-release stretch-backports install -y golang-go \
+&& apt install -y \
+git \
+curl \
+build-essential \
+libseccomp-dev \
+btrfs-tools \
+libdevmapper-dev \
+musl-tools \
+util-linux \
+ca-certificates
+
+
+
+
+# Build firecracker itself
+# TODO can we cache cargo artifacts like we do go modules?
+FROM build-base as firecracker-build
+SHELL ["/bin/bash", "-c"]
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain 1.32.0 \
```

Comment:
Yeah makes sense, I avoided the rust base image initially just because I wanted to make 100% sure all the builds were using the same set of dependencies (that we apt install in `build-base`).

It turns out [the rust dockerfile](https://github.com/rust-lang/docker-rust/blob/edda09581f7c015c5cc2b8f0e04951a83b6f9727/1.32.0/stretch/Dockerfile) happens to also use `debian:stretch` as a base and works by downloading `rust-init` and verifying it against hardcoded SHA sums. I'll switch to using that base instead for the firecracker install.

Reply:
Using the Rust base image ended up causing complication with wanting to run the builds as a non-root user, so I just took a few lines from their dockerfile (download rust-init and verify its sha sum) and put it in here.